server/app/api.py

```diff
@@ -1071,7 +1040,34 @@ def download(self, obj, user, data):
         :param data: (dictionary) data
         :return: file contents in utf-8
         """
-        return self.make_zip_response(*self.zip(obj, user, data))
+        # Create a fake file object for our response
+        class ResponseFile(object):
+            def __init__(self, write):
+                self.bytes_written = 0
+                self._write = write
+
+            def write(self, bytes):
+                self.bytes_written += len(bytes)
+                self._write(bytes)
+
+            def tell(self):
+                return self.bytes_written
+
+            def flush(self):
+                pass
+
+        # WGSI app to get a write() function. See PEP 333
+        def zip_response(environ, start_response):
+            name, files = self.data_for_zip(obj)
+            write = start_response('200 OK', [
+                ('Content-Disposition', 'attachment; filename=submission-%s.zip' % name),
+                ('Content-Type', 'application/zip')
+            ])
+            stream = ResponseFile(write)
+            with zf.ZipFile(stream, 'w') as zipfile:
+                add_to_zip(zipfile, files)
+            return ''
+        return make_response(zip_response)
```

Comment:
I think it streams: https://www.python.org/dev/peps/pep-0333/#the-write-callable is under the section "Buffering and Streaming"